packages/modular-scripts/src/esbuild-scripts/start/utils/generateSelfSignedCert.ts

```diff
@@ -0,0 +1,144 @@
+import fs from 'fs';
+import path from 'path';
+import selfsigned from 'selfsigned';
+import { promisify } from 'util';
+import rimraf from 'rimraf';
+
+import * as logger from '../../../utils/logger';
+
+const del = promisify(rimraf);
+
+// This code is mostly duplicated from webpack-dev-server, with some typings added.
+// To provide consistent https behaviour in esbuild mode, we use the same directories to store the generated cert.
+// See: https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js
+
+function findCacheDir() {
+  const cwd = process.cwd();
+
+  let dir: string | undefined = cwd;
+
+  for (;;) {
```

Comment:
This loop finds the first parent package.json - I believe we already use `find-up` for this in other parts of the code -  can we do that here as well for simplicity / readability? That infinite loop looks scary.

Reply:
As per the comment near the top - this file is >85% duplicated from webpack-dev-server: https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L418

The only changes I'm making are

- Types
- Necessary material changes that allow us to achieve the goal of this PR

The infinite loop here is just duplicated. We can diverge and use our own implementation, but is that we want?

Reply:
Sounds like we can de-duplicate some of this since my PR Types webpack-dev-server - and maybe improve this logic too